anvil/src/shell.rs

```diff
@@ -671,31 +671,32 @@ impl<BackendData: Backend> XdgShellHandler for AnvilState<BackendData> {
 
     fn grab(&mut self, surface: PopupSurface, seat: wl_seat::WlSeat, serial: Serial) {
         let seat: Seat<AnvilState<BackendData>> = Seat::from_resource(&seat).unwrap();
-        let ret = self
-            .popups
-            .grab_popup(&self.display_handle, surface.wl_surface().clone(), &seat, serial);
-
-        if let Ok(mut grab) = ret {
-            if let Some(keyboard) = seat.get_keyboard() {
-                if keyboard.is_grabbed()
-                    && !(keyboard.has_grab(serial)
-                        || keyboard.has_grab(grab.previous_serial().unwrap_or(serial)))
-                {
-                    grab.ungrab(PopupUngrabStrategy::All);
-                    return;
+        let kind = PopupKind::Xdg(surface);
```

Comment:
We could, though I appreciate the abstraction it provides, in case we ever get another popup like type. (Maybe this makes sense for Xwayland?)

Anyway, is this a concern for this PR?

Reply:
Not really a concern, just a small thing I noticed.

Reply:
We could definitely track this in another issue and have a proper discussion on it. Right now that type is not providing anything.